### PR TITLE
[docs] - Clean up Dagster+ Branch Deployment (CI) docs

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR cleans up the Branch Deployment docs.

### CODE REVIEW

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx [hunk 1]
```diff
@@ -30,87 +33,108 @@ To complete the steps in this guide, you'll need:
 
 ## Step 1: Generate a Dagster+ agent token
```
Comment (erinkcochran87) [inline docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:34]: Do we still recommend creating a dedicated agent for this for Hybrid?

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx [hunk 2]
```diff
@@ -30,87 +33,108 @@ To complete the steps in this guide, you'll need:
...
-In this step, you'll add the required GitHub workflow files to your GitHub repository. The files can be found in our [Hybrid quickstart repository](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart) or [Serverless quickstart repository](https://github.com/dagster-io/dagster-cloud-serverless-quickstart), depending on the agent you are using.
+In this step, you'll add the required GitHub Action files to your GitHub repository. The files can be found in the [Hybrid quickstart repository](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart) or [Serverless quickstart repository](https://github.com/dagster-io/dagster-cloud-serverless-quickstart), depending on the deployment type you're using.
 
-Copy the following files to your repository (the linked files are shown in the Hybrid repo, there are equivalents in the Serverless repo).
+<TabGroup>
+<TabItem name="Hybrid">
 
-- [`dagster_cloud.yaml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/dagster_cloud.yaml)
-- [`.github/workflows/dagster-cloud-deploy.yml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/.github/workflows/dagster-cloud-deploy.yml)
+### Hybrid
```
Comment (erinkcochran87) [inline docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:67]: TLDR; at the end

I find the UI experience of adding a code location in Hybrid kinda confusing. We ask users to add the code location in a YAML editor, which contains the same properties as a `dagster_cloud.yaml`. Since there isn't a connected repository, specifying this in Dagster+ doesn't automatically create/add a `dagster_cloud.yaml` to the project.

So a user then has to manually create the `dagster_cloud.yaml` themselves with the same info they specify using the YAML editor - right? 

I haven't tested this, but if someone tries to add a code location/project using the CLI but there isn't a `dagster_cloud.yaml`, what happens?

TLDR: This feels like a weird experience and it's proving difficult to describe here and in other guides.

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx [hunk 3]
```diff
@@ -30,87 +33,108 @@ To complete the steps in this guide, you'll need:
...
+             aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+             aws-region: ${{ secrets.AWS_REGION }}
+   ```
+
+4. Save and commit the files to the repository.
+
+</TabItem>
+<TabItem name="Serverless">
+
+### Serverless
```
Comment (erinkcochran87) [inline docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:109]: I don't believe the user needs to change anything in these files if they're using Serverless, but confirmation would be good

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx [hunk 4]
```diff
@@ -30,87 +33,108 @@ To complete the steps in this guide, you'll need:
...
+  <strong>If you used the Dagster+ GitHub app to import your project</strong>,
+  this step was already done for you.{" "}
+  <a href="#step-5-access-the-branch-deployment">Skip to the next step</a>.
 </Note>
 
+In this step, you'll add the required secrets to the GitHub repository. **Note**: This isn't how environment variables and secrets can be used in Dagster code. Refer to the [Environment variables and secrets guide](/dagster-plus/managing-deployments/environment-variables-and-secrets) for more information.
+
 <AddGitubRepositorySecret />
 
 **For Hybrid deployments,** repeat steps 3-6 for each of the secrets required for your registry type:
```
Comment (erinkcochran87) [inline docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:140]: For Serverless, users should only need to set `DAGSTER_CLOUD_API_TOKEN` AND `ORGANIZATION_ID`, correct?

**Comment by erinkcochran87** [comment]
@jamiedemaria This is the Branch Deps content I had questions about, which I left as comments. I plan to use it as a model when updating the Gitlab version of this guide - thanks for taking a look!

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 1 above)
Comment (jamiedemaria) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:34]: I don't know the answer to this. My guess is yes, since none of the work I did recently for branch deployments affected agents, but maybe there was some other work that has happened that i'm unaware of. Maybe @gibsondan will know

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 3 above)
Comment (jamiedemaria) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:109]: The only thing I see that they could maybe want to change is uncommenting these lines to pass env vars 

https://github.com/dagster-io/dagster-cloud-serverless-quickstart/blob/main/.github/workflows/deploy.yml#L76

https://github.com/dagster-io/dagster-cloud-serverless-quickstart/blob/main/.github/workflows/branch_deployments.yml#L74

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx [hunk 5]
```diff
@@ -30,87 +33,108 @@ To complete the steps in this guide, you'll need:
...
 
-In the next step, you'll modify these files to work with your Dagster+ setup.
+1. Add the following files to the repository:
 
----
+   - [`dagster_cloud.yaml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/dagster_cloud.yaml)
+   - [`.github/workflows/dagster-cloud-deploy.yml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/.github/workflows/dagster-cloud-deploy.yml)
 
-## Step 4: Configure the GitHub repository
+2. In the [`dagster_cloud.yaml` file](/dagster-plus/managing-deployments/dagster-cloud-yaml), replace `build.registry` with the registry used by the [agent you created in Step 2](#step-2-create-and-configure-an-agent). For example:
```
Comment (jamiedemaria) [inline docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:74]: Probably missing something very basic here, but I'm a bit confused by adding the registry here. If this file is getting committed, wouldnt the agent specified in `build.registry` also be used when deployed to prod? Even through we created the agent with the intent for it to be used for branch deployments

Since this section has existed since this guide was first written, this question isn't blocking in any way, more just for my own knowledge

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 2 above)
Comment (jamiedemaria) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:67]: hmmm I don't have a compete answer for this either. 

One thing that could help is that in this page https://docs.dagster.io/dagster-plus/managing-deployments/code-locations we say that using a `dagster_cloud.yaml` file is recommended for code locations in the first section, but then there isn't a section later in the page that describes adding/modifying/deleting code locations when you're using `dagster_cloud.yaml`

**Review by jamiedemaria — APPROVED** [review]
this looks good to me. I think @gibsondan  would be able to spot check the questions about making an agent and code locations

**Comment by erinkcochran87** [comment]
@gibsondan Jamie suggested you might be able to help us answer some of the questions on this PR, so I added you as a reviewer to get this to show up in your queue. Would you mind taking a look? :)

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 3 above)
Comment (erinkcochran87) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:109]: I know there've been questions about these parts of the files before, but I can't recall the what/why. Why do users need to uncomment these lines if they're using our recommended method of env var management in the UI?

**Comment by gibsondan** [comment]
I can take a look, sure - this actually does not show up in my graphite queue though since jamie already accepted the PR

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 1 above)
Comment (gibsondan) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:34]: I think so yes, it will make it less likely that your branch deployments eat up all your resources for your prod cluster

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 2 above)
Comment (gibsondan) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:67]: I think we should steer away from recommending people use the UI to add a code location, for the reason that you mention, especially in a guide about CI/CD. They shouldn't need to do that at all for this to work.

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 5 above)
Comment (gibsondan) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:74]: prod and branch deployments use different dagster_cloud.yaml files, so it has to be specfified in both. I could imagine users using a different docker registry for their branch deployment images vs. their prod ones

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 3 above)
Comment (gibsondan) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:109]: They don't need to comment that out if they are setting env vars in the UI - that part was for the old way of setting secrets where the values were included in the Docker image, the new way is better than that from a security standpoint

**Review by gibsondan — APPROVED** [review]

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 3 above)
Comment (erinkcochran87) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:109]: That's right! Thanks for the refresher, Dan!

File: docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx (hunk 5 above)
Comment (erinkcochran87) [reply docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx:74]: lol now I feel like I'm missing something. This is how I'm now interpreting this - is this correct?

- There's a `dagster_cloud.yaml`, set via the UI, that imports the project and allows the `prod` deployment to work
- There's another `dagster_cloud.yaml` in the GH repository that sets the agent registry and allows Branch Deps to work

**Comment by gibsondan** [comment]
when you say “set via the UI” - which screen are you referring to? The NUX
when you add a new code location?

**Comment by erinkcochran87** [comment]
Either the NUX for a brand new account, or using the **Code locations** tab to add a new project to an existing account. I believe the process for this in Hybrid is the same regardless of the state of onboarding.

**Comment by gibsondan** [comment]
You don’t need to use the UI at all to add a new code location (in either
serverless or hybrid). If you create a new GitHub repo and follow the steps
in the guide, the GitHub action will create the new code location for you.

In serverless, the NUX automates the creation of a new GitHub repository if
you want, but that is optional.

**Comment by PedramNavid** [comment]
Closing as there's no owner on this PR and it has grown stale.
